build_id_getter.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"path/filepath"
+
+	"github.com/bitrise-steplib/bitrise-step-artifact-pull/model"
+)
+
+const DELIMITER = "."
+
+type BuildIDGetter interface {
+	GetBuildIDs() ([]string, error)
+}
+
+type DefaultBuildIDGetter struct {
+	FinishedStages model.FinishedStages
+	TargetNames    []string
+}
+
+func NewDefaultBuildIDGetter(finishedStages model.FinishedStages, targetNames []string) BuildIDGetter {
+	return DefaultBuildIDGetter{
+		FinishedStages: finishedStages,
+		TargetNames:    targetNames,
+	}
+}
+
+func (bg DefaultBuildIDGetter) GetBuildIDs() ([]string, error) {
+	var buildIDs []string
+
+	stageWorkflowMap := bg.createWorkflowMap()
+
+	for _, target := range bg.TargetNames {
+		for k, v := range stageWorkflowMap {
+			matched, err := filepath.Match(target, k)
```

Comment:
This will only work if the delimiter does not change ever. Wouldn't it be better here to explicitly write out the wildcard cases?